syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -450,3 +451,84 @@ def _restore_from_state(self, state: dict):
 
     def set_random_state(self, random_state: np.random.RandomState):
         self.random_state = random_state
+
+
+class SearcherWithRandomSeedAndFilterDuplicates(SearcherWithRandomSeed):
```

Comment:
This is code to be introduced in #487, which will be merged before this one here.